packages/rum-core/src/performance-monitoring/transaction-service.js

```diff
@@ -139,28 +139,36 @@ class TransactionService {
        * allow a redefinition until there's a call that doesn't have that
        * or the threshold is exceeded.
        */
+      let redefineType = tr.type
+      const currentTypeOrder = TRANSACTION_TYPE_ORDER.indexOf(tr.type)
+      const redefineTypeOrder = TRANSACTION_TYPE_ORDER.indexOf(type)
+
+      if (currentTypeOrder >= 0) {
+        /**
+         * If the redefined type is not present in the predefined order, that implies
+         * it's a user defined type and it is of higher precedence
+         */
+        if (redefineTypeOrder === -1) {
+          redefineType = type
+        } else if (
+          /**
+           * Update type based on precedence defined in TRANSACTION_TYPE_ORDER.
+           * If either orders don't exist we also don't redefine the type.
+           */
+          redefineTypeOrder >= 0 &&
+          redefineTypeOrder < currentTypeOrder
+        ) {
+          redefineType = type
+        }
```

Comment:
I think this if can be written as follows and it would have the same effect:

```suggestion
        if (redefineTypeOrder < currentTypeOrder) {
          redefineType = type
        }
```

Reply:
try harding `if else` 😅